File: src/main/java/fr/epicanard/mapsaver/models/map/PlayerMap.java
package fr.epicanard.mapsaver.models.map;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.UUID;

@NoArgsConstructor
@AllArgsConstructor
@Getter
public class PlayerMap {
    private UUID playerUuid;
    private UUID mapUuid;
    private boolean owner;
    private Visibility visibility;
    private String name;
}

File: src/main/scala/fr/epicanard/mapsaver/models/map/PlayerMap.scala
package fr.epicanard.mapsaver.models.map

import java.util.UUID

case class PlayerMap(
    playerUuid: UUID,
    dataId: Int,
    owner: Boolean,
    visibility: Visibility,
    name: String,
    locked: Boolean
)

object PlayerMap {

  def tupled: ((UUID, Int, Boolean, Visibility, String, Boolean)) => PlayerMap = (PlayerMap.apply _).tupled

  def fromMapToSave(mapToSave: MapToSave, dataId: Int) =
    new PlayerMap(mapToSave.owner, dataId, true, mapToSave.visibility, mapToSave.name, false)
}
